odl/tomo/geometry/detector.py

```diff
@@ -51,18 +51,18 @@ class Detector(with_metaclass(ABCMeta, object)):
     * optionally a sampling grid for the parameters
     """
 
-    def __init__(self, params, grid=None, ndim=None):
+    def __init__(self, ndim, params, grid=None):
         """Initialize a new instance.
 
         Parameters
         ----------
+        ndim : `int` in (0, 1 or 2)
```

Comment:
Why any restriction? Who says that 3- or 4-dimensional detectors are impossible? I'd just say `nonnegative int`.